kb/0003-scripting-functions.md

```diff
@@ -5,7 +5,711 @@ _[back to Index](../index.md)_
 
 **Related articles**: [Scripting Language Overview](0002-scripting-overview.md)
 
-Here you will find the list of the built-in and some other useful functions for the game scripting.
+Here you will find the list of the built-in and some game-level useful functions for the game scripting.
+
+
+## Built-In Functions
+
+Below are the functions which make part of the compiler API. Each function has its own unique identifier token that's also listed below.
+
+### Validation
+
+* **GetEngineVersion**: return `ENGINE_SCRIPT_VERSION` constant (defined in `core.h`). 
+    * **Compiler Token**: `FUNC_GETENGINEVERSION`
+    * The engine version is pushed back to the compiler, and an error will be produced in case of version mismatch. Therefore, retrieving and using it in the scripting language after the initial call isn't necessary.
+    ``` C++
+    syntax:
+        int GetEngineVersion();
+    usage: 
+        GetEngineVersion();
+    ```
+* **CheckFunction**: verify if a function with specified name exists. 
+    * **Compiler Token**: `FUNC_CHECKFUNCTION`
+    * `value`: function name.
+    ``` C++
+    syntax:
+        bool CheckFunction(string value);
+    usage: 
+        if (CheckFunction("ControlsTreeInit"))
+        {
+            ControlsTreeInit();
+        }
+    ```
+* **TestRef**: 
+    * **Compiler Token**: `FUNC_TEST_REF`
+    ``` C++
+    syntax:
+        int TestRef(1); 
+    ```
+
+### Utility
+
+* **SetTimeScale**: Set the speed of the game simulation for the whole engine. For your epic slo-mo moments.
+    * **Compiler Token**: `FUNC_SETTIMESCALE`
+    * `value`: time scale, from 0 to 1.
+    * Internally, time scale is stored as a `float`, so passing both `float` and `int` values is allowed. 
+    * Because the value affects the whole game, make sure you don't forget to return the time scale back to normal! 
+    ``` C++
+    syntax:
+        void SetTimeScale(int value);
+        void SetTimeScale(float value);
+    usage: 
+        SetTimeScale(0.0);
+    ```
+
+NOTE(yakvi): Below is work-in-progress
```

Comment:
You want to leave it as a reminder for yourself?

Reply:
Yes, and for the readers to indicate that below is not complete.